server/v2/cometbft/abci.go

```diff
@@ -237,6 +237,7 @@ func (c *Consensus[T]) InitChain(ctx context.Context, req *abciproto.InitChainRe
 
 	// store chainID to be used later on in execution
 	c.chainID = req.ChainId
+	c.cfg.InitialHeight = uint64(req.InitialHeight) // TODO: cfg is loaded initially ???
```

Comment:
cc @julienrbrt also we should document why this is being set in a small godoc so the next person understands the reasoning

Reply:
What does this comment means?